src/Caching/SimpleStringCache.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pelago\Emogrifier\Caching;
+
+/**
+ * This cache caches string values with string keys. It is not PSR-6-compliant.
+ *
+ * Usage:
+ *
+ * ```php
+ * $cache = new SimpleStringCache();
+ * $cache->set($key, $value);
+ * …
+ * if ($cache->has($key) {
+ *     $cachedValue = $cache->get($value);
+ * }
+ * ```
+ */
+class SimpleStringCache
+{
+    /**
+     * @var array<string, string>
+     */
+    private $values = [];
+
+    /**
+     * Checks whether there is an entry stored for the given key.
+     *
+     * @param string $key the key to check; must not be empty
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function has(string $key): bool
+    {
+        $this->checkForNonEmptyKey($key);
+
+        return isset($this->values[$key]);
+    }
+
+    /**
+     * Returns the entry stored for the given key, and throws an exception if the value does not exist
+     * (which helps keep the return type simple).
+     *
+     * @param string $key the key to of the item to retrieve; must not be empty
+     *
+     * @return string the retrieved value; may be empty
+     *
+     * @throws \BadMethodCallException
+     */
+    public function get(string $key): string
+    {
+        if (!$this->has($key)) {
+            throw new \BadMethodCallException('You can only call `get` with a key for an existing value.', 1625996246);
+        }
+
+        return $this->values[$key];
+    }
+
+    /**
+     * Sets or overwrites an entry.
+     *
+     * @param string $key the key to of the item to set; must not be empty
+     * @param string $value the value to set; can be empty
+     *
+     * @throws \BadMethodCallException
+     */
+    public function set(string $key, string $value): void
+    {
+        $this->checkForNonEmptyKey($key);
+
+        $this->values[$key] = $value;
+    }
+
+    /**
+     * @throws \InvalidArgumentException
+     */
+    private function checkForNonEmptyKey(string $key): void
```

Comment:
Wondering about the method name.  Would something like `assertNotEmptyKey` be more descriptive?  If `assert` may sound like conflating with test methods, perhaps `verify` or `confirm` (or something like) - these verbs all imply that an empty key will be rejected, whereas `checkFor` doesn't really say what will be done when the key is/isn't found to be empty.